oracle_cardano/processor/tx_processors/success/bridging_requested_processor_skyline.go

```diff
@@ -0,0 +1,353 @@
+
+		p.logger.Info("Added RefundRequestClaim",
+		"txHash", tx.Hash, "metadata", metadata, "claim", core.RefundRequestClaimString(claim))
+}
+*/
+
+func (p *BridgingRequestedProcessorSkylineImpl) validate(
+	tx *core.CardanoTx, metadata *common.BridgingRequestMetadata, appConfig *cCore.AppConfig,
+) error {
+	multisigUtxo, err := utils.ValidateTxOutputs(tx, appConfig, false)
+	if err != nil {
+		return err
+	}
+
+	cardanoSrcConfig, _ := cUtils.GetChainConfig(appConfig, tx.OriginChainID)
+	if cardanoSrcConfig == nil {
+		return fmt.Errorf("unsupported chain id found in tx. chain id: %v", tx.OriginChainID)
+	}
+
+	cardanoDestConfig, _ := cUtils.GetChainConfig(appConfig, metadata.DestinationChainID)
+	if cardanoDestConfig == nil {
+		return fmt.Errorf("destination chain not registered: %v", metadata.DestinationChainID)
+	}
+
+	if len(metadata.Transactions) > appConfig.BridgingSettings.MaxReceiversPerBridgingRequest {
+		return fmt.Errorf("number of receivers in metadata greater than maximum allowed - no: %v, max: %v, metadata: %v",
+			len(metadata.Transactions), appConfig.BridgingSettings.MaxReceiversPerBridgingRequest, metadata)
+	}
+
+	nativeCurrencyAmountSum := big.NewInt(0)
+	wrappedTokenAmountSum := big.NewInt(0)
+
+	feeSum := uint64(0)
+	foundAUtxoValueBelowMinimumValue := false
+	foundAnInvalidReceiverAddr := false
+
+	exchangeMargin := float64(0.000001)
+
+	exchangeRate, err := GetExchangeRate(tx.OriginChainID, metadata.DestinationChainID)
+	if err != nil {
+		return err
+	}
+
+	for _, receiver := range metadata.Transactions {
+		receiverAddr := strings.Join(receiver.Address, "")
+
+		if !cardanotx.IsValidOutputAddress(receiverAddr, cardanoDestConfig.NetworkID) {
+			foundAnInvalidReceiverAddr = true
+
+			break
+		}
+
+		var (
+			totalNativeCurrencyAmountToBridge uint64
+			totalWrappedTokenAmountToBridge   uint64
+		)
+
+		totalWrappedTokenAmountToBridge = uint64(0)
+
+		if receiver.Additional == nil {
+			receiver.Additional = &common.BridgingRequestMetadataCurrencyInfo{}
+		}
+
+		if receiver.IsNativeTokenOnSrc {
+			// amount_to_bridge must be >= minUtxoAmount on destination
+			if receiver.Amount < cardanoDestConfig.UtxoMinAmount {
+				foundAUtxoValueBelowMinimumValue = true
+
+				break
+			}
+
+			totalNativeCurrencyAmountToBridge = receiver.Additional.SrcAmount
+			totalWrappedTokenAmountToBridge = receiver.Amount
+		} else {
+			// amount_to_bridge must be >= minUtxoAmount on source
+			if receiver.Amount < cardanoSrcConfig.UtxoMinAmount {
+				foundAUtxoValueBelowMinimumValue = true
+
+				break
+			}
+
+			if receiver.Additional.DestAmount < cardanoDestConfig.UtxoMinAmount {
+				foundAUtxoValueBelowMinimumValue = true
+
+				break
+			}
+
+			totalNativeCurrencyAmountToBridge = receiver.Amount + receiver.Additional.SrcAmount
+		}
+
+		exchangeDiff := float64(receiver.Additional.DestAmount)*exchangeRate - float64(receiver.Additional.SrcAmount)
+		if (exchangeDiff > 0 && exchangeDiff <= exchangeMargin) ||
```

Comment:
shouldnt it be the inverse?
if (exchangeDiff > 0 && exchangeDiff > exchangeMargin) || (exchangeDiff < 0 && exchangeDiff < -1*exchangeMargin)
then error?